config/routes.rb

```diff
@@ -5,4 +5,5 @@
   resources :books, only: [:index, :show]
   devise_for :users, controllers: { registrations: "registrations" }
   resources :pending_purchases, except: [:new]
+  resources :charges
```

Comment:
I'm not sure, but I don't think you're using all of the endpoints (index, new, create, show, edit, update, delete). You should disable whichever ones you're not using.